pallets/registrar/src/weights.rs

```diff
@@ -55,247 +63,301 @@ pub trait WeightInfo {
+	/// Proof: `Registrar::PendingPaused` (`max_values`: Some(1), `max_size`: None, mode: `Measured`)
+	/// Storage: `Registrar::Paused` (r:1 w:0)
+	/// Proof: `Registrar::Paused` (`max_values`: Some(1), `max_size`: None, mode: `Measured`)
+	/// Storage: `Session::CurrentIndex` (r:1 w:0)
+	/// Proof: `Session::CurrentIndex` (`max_values`: Some(1), `max_size`: None, mode: `Measured`)
+	/// Storage: `Registrar::PendingToRemove` (r:1 w:1)
+	/// Proof: `Registrar::PendingToRemove` (`max_values`: Some(1), `max_size`: None, mode: `Measured`)
+	/// The range of component `x` is `[5, 3000000]`.
+	/// The range of component `y` is `[1, 50]`.
+	fn deregister_scheduled(x: u32, y: u32, ) -> Weight {
+		// Proof Size summary in bytes:
+		//  Measured:  `454 + y * (4 ±0)`
+		//  Estimated: `1937 + y * (4 ±0)`
+		// Minimum execution time: 25_398_000 picoseconds.
+		Weight::from_parts(32_217_636, 1937)
+			// Standard Error: 0
+			.saturating_add(Weight::from_parts(2, 0).saturating_mul(x.into()))
+			// Standard Error: 8_573
+			.saturating_add(Weight::from_parts(242_112, 0).saturating_mul(y.into()))
+			.saturating_add(T::DbWeight::get().reads(6_u64))
+			.saturating_add(T::DbWeight::get().writes(2_u64))
+			.saturating_add(Weight::from_parts(0, 4).saturating_mul(y.into()))
+	}
+	/// Storage: `Registrar::PendingVerification` (r:1 w:1)
+	/// Proof: `Registrar::PendingVerification` (`max_values`: Some(1), `max_size`: None, mode: `Measured`)
+	/// Storage: `Registrar::PendingParaIds` (r:1 w:1)
+	/// Proof: `Registrar::PendingParaIds` (`max_values`: Some(1), `max_size`: None, mode: `Measured`)
+	/// Storage: `Registrar::PendingPaused` (r:1 w:0)
+	/// Proof: `Registrar::PendingPaused` (`max_values`: Some(1), `max_size`: None, mode: `Measured`)
+	/// Storage: `Registrar::RegisteredParaIds` (r:1 w:0)
+	/// Proof: `Registrar::RegisteredParaIds` (`max_values`: Some(1), `max_size`: None, mode: `Measured`)
+	/// Storage: `Registrar::Paused` (r:1 w:0)
+	/// Proof: `Registrar::Paused` (`max_values`: Some(1), `max_size`: None, mode: `Measured`)
+	/// Storage: `Session::CurrentIndex` (r:1 w:0)
+	/// Proof: `Session::CurrentIndex` (`max_values`: Some(1), `max_size`: None, mode: `Measured`)
 	/// The range of component `y` is `[1, 50]`.
 	fn mark_valid_for_collating(y: u32, ) -> Weight {
 		// Proof Size summary in bytes:
-		//  Measured:  `418 + y * (8 ±0)`
-		//  Estimated: `1899 + y * (8 ±0)`
-		// Minimum execution time: 17_420_000 picoseconds.
-		Weight::from_parts(32_834_209, 1899)
-			// Standard Error: 16_866
-			.saturating_add(Weight::from_parts(391_115, 0).saturating_mul(y.into()))
-			.saturating_add(T::DbWeight::get().reads(4_u64))
+		//  Measured:  `477 + y * (8 ±0)`
+		//  Estimated: `1958 + y * (8 ±0)`
+		// Minimum execution time: 27_650_000 picoseconds.
+		Weight::from_parts(42_754_882, 1958)
+			// Standard Error: 9_207
+			.saturating_add(Weight::from_parts(276_708, 0).saturating_mul(y.into()))
+			.saturating_add(T::DbWeight::get().reads(6_u64))
```

Comment:
There are 2 reads more here because we read Paused and PendingPaused, which is not needed. I will add back the function `schedule_parachain_change` and use it in `mark_valid_for_collating`.